src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Text_ExcelCompat.txt

```diff
@@ -2824,4 +2824,456 @@ Error({Kind:ErrorKind.InvalidArgument})
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-###k#0#####0"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-####0####0###k#"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-##k##0####0###k#"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-####0##k##0####"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-#\##0#####0"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-###\##00"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-###\##0#####0"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-####0####0###\##"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-##\###0####0###\##"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "0.00e-####0##\###0####"}, Text(1234567.1234567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "(0.00e-0k#);(0.00e-#k##0##0##)"}, Text(12345.34567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "##;(0.00e-###0#k#0##)"}, Text(-12345.34567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "(0.00e-0#);##;(0.00e-#####0####0###k#0###)"}, Text(0, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "(0.00e-####0####);(0.00e-####0####0#k#)"}, Text(12345.34567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "(0.00e-####0######);(0.00e-####0####0#k#)"}, Text(-12345.34567, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "(0.00e-0#);(0.00e-0##k##0##);(0.00e-####0###k##0##k##0###)"}, Text(0, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+>> With({fmt: "(0.00E+0#);(0.00e-0##""k""##0##);(0.00E+####0###k##0##""k""##0###)"}, Text(0, fmt))
+Error({Kind:ErrorKind.InvalidArgument})
+
+// # after exponential notation
+>> Text(12345.34567,"0.00E+#####00")
+"1.23E+04"
+
+>> Text(12345.34567,"0.00E+#00")
+"1.23E+04"
+
+>> Text(12345.34567,"0.00E+#####0#0")
+"1.23E+04"
+
+>> Text(12345.34567,"0.00E+#0#0")
+"1.23E+04"
+
+>> Text(12345.34567,"0.00E+#####0####0")
+"1.23E+04"
+
+>> Text(12345.34567,"0.00E+#0#####0")
+"1.23E+04"
+
+>> Text(12345.34567,"0.00E+00#####")
+"1.23E+004"
+
+>> Text(12345.34567,"0.00E+#####00#")
+"1.23E+004"
+
+>> Text(12345.34567,"0.00E+#####00#####")
+"1.23E+004"
+
+>> Text(12345.34567,"0.00E+0#0#####")
+"1.23E+004"
+
+>> Text(12345.34567,"0.00E+0####0#####")
+"1.23E+004"
+
+>> Text(12345.34567,"0.00E+####0####0#####")
+"1.23E+004"
+
+>> Text(12345.34567,"(0.00E+00##);##")
+"(1.23E+004)"
+
+>> Text(-12345.34567,"(0.00E+00##);##")
+"12345"
+
+>> Text(12345.34567,"##;(0.00E+00##)")
+"12345"
+
+>> Text(-12345.34567,"##;(0.00E+00##)")
+"(1.23E+004)"
+
+>> Text(0,"(0.00E+0#);##;(0.00E+000###)")
+"(0.00E+0000)"
```

Comment:
>0000 [](http://example.com/codeflow?start=8&length=4)

Why is it 4 zeros? The format only specifies 3 required places, and the value (0) is not larger than that.

Reply:
Same issue in many other test cases

Reply:
This is how excel does. # after 0 count as another 0. Can you recheck ?

Reply:
Oh yeah, Excel is weird :-|... I guess that's right then :-)